app/models/user.rb

```diff
@@ -4,4 +4,30 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+
+  def has_membership_application?
+    membership_applications.size > 0 && (membership_applications.select { |app| app.persisted? }).count > 0
```

Comment:
That makes sense to me.  (I can't think of a case where something would not have been persisted.  In the case of invalid data, there'd be a ROLLBACK)

@thesuss, you've been doing this -- testing for persistence -- in your checks.  Do we really need to keep doing that?